runtime/src/bank.rs

```diff
@@ -1791,6 +1792,21 @@ pub fn goto_end_of_slot(bank: &mut Bank) {
     }
 }
 
+// This protects from memory exhaustion.
+// Because we're are about to restore full process state from file, allow use
+// of memory as much as possible. But set a hard-limit with ample of head rooms.
+// This is an extra protection for possible bincode bugs in addition to
+// deserialize_snapshot_data_file
+pub fn deserialize_for_snapshot<R, T>(reader: R) -> bincode::Result<T>
```

Comment:
How about this name:
```suggestion
pub fn deserialize_from_snapshot<R, T>(reader: R) -> bincode::Result<T>
```

Reply:
Nice!: 5c8a4003405a35196b9d4c6535f38bbe653adbce